packages/react/src/JsonFormsContext.tsx

```diff
@@ -205,107 +210,219 @@ export const ctxToCellProps = (
+    ({ ctx, props }: JsonFormsStateContext & ArrayLayoutProps) => {
+      const arrayLayoutProps = ctxToArrayLayoutProps(ctx, props);
+      const dispatchProps = ctxDispatchToArrayControlProps(ctx.dispatch);
+      return (<Component {...arrayLayoutProps} {...dispatchProps} />);
+    };
+
+export const withContextToArrayControlProps =
+  (Component: ComponentType<ArrayControlProps>): ComponentType<OwnPropsOfControl> =>
+    ({ ctx, props }: JsonFormsStateContext & ArrayControlProps) => {
+      const stateProps = ctxToArrayControlProps(ctx, props);
+      const dispatchProps = ctxDispatchToArrayControlProps(ctx.dispatch);
+
+      return (<Component {...props} {...stateProps} {...dispatchProps} />);
+    };
+
+export const withContextToMasterListItemProps =
+  (Component: ComponentType<StatePropsOfMasterItem>): ComponentType<OwnPropsOfMasterListItem> =>
+    ({ ctx, props }: JsonFormsStateContext & StatePropsOfMasterItem) => {
+      const stateProps = ctxToMasterListItemProps(ctx, props);
+      return (<Component {...stateProps} />);
+    };
+
+export const withContextToCellProps =
+  (Component: ComponentType<CellProps>): ComponentType<OwnPropsOfCell> =>
+    ({ ctx, props }: JsonFormsStateContext & CellProps) => {
+      const cellProps = ctxToCellProps(ctx, props);
+      const dispatchProps = ctxDispatchToControlProps(ctx.dispatch)
+
+      return (<Component {...props} {...dispatchProps} {...cellProps} />);
+    };
+
+export const withContextToEnumCellProps =
+  (Component: ComponentType<EnumCellProps>): ComponentType<OwnPropsOfEnumCell> =>
+    ({ ctx, props }: JsonFormsStateContext & OwnPropsOfEnumCell) => {
+      const cellProps = ctxToCellProps(ctx, props);
+      const dispatchProps = ctxDispatchToControlProps(ctx.dispatch);
+      const options =
+        props.options !== undefined ? props.options : props.schema.enum;
+
+      return (<Component {...props} {...dispatchProps} {...cellProps} options={options} />);
+    };
+
+export const withContextToEnumProps =
+  (Component: ComponentType<ControlProps & OwnPropsOfEnum>): ComponentType<OwnPropsOfControl & OwnPropsOfEnum> =>
+    ({ ctx, props }: JsonFormsStateContext & ControlProps & OwnPropsOfEnum) => {
+      const stateProps = ctxToControlProps(ctx, props);
+      const dispatchProps = ctxDispatchToControlProps(ctx.dispatch);
+      const options =
+        props.options !== undefined ? props.options : stateProps.schema.enum;
+
+      return (<Component {...props} {...dispatchProps} {...stateProps} options={options} />);
+    };
+
+// --
+
+type JsonFormsPropTypes = ControlProps | CombinatorProps | LayoutProps | CellProps;
+
+export const areEqual = (prevProps: JsonFormsPropTypes, nextProps: JsonFormsPropTypes) => {
+  const prev = omit(prevProps, ['handleChange', 'renderers', 'cells', 'uischemas']);
+  const next = omit(nextProps, ['handleChange', 'renderers', 'cells', 'uischemas']);
+  return isEqual(prev, next)
```

Comment:
shouldn't this be:
`isEqual && renderers.length==renderers.length && cells.length === cells.length && uischemas.length === uischemas.length`

Reply:
Right I somehow managed to switch the logic in between, thanks for noticing